libsolidity/interface/StandardCompiler.cpp

```diff
@@ -117,7 +117,7 @@ bool hashMatchesContent(string const& _hash, string const& _content)
 	{
 		return dev::h256(_hash) == dev::keccak256(_content);
 	}
-	catch (dev::BadHexCharacter)
+	catch (...)
```

Comment:
Can you show the appropriate gcc warning/error you've had?

Reply:
The error is `Werror=catch-value`.
In case a different exception is thrown (at least in one of the places I removed the explicite exceptino `dev::BadHexCharacter`, no value would've been returned.

```bash
/mnt/data/repos/solidity/libsolidity/interface/StandardCompiler.cpp: In function ‘bool {anonymous}::hashMatchesContent(const string&, const string&)’:
/mnt/data/repos/solidity/libsolidity/interface/StandardCompiler.cpp:120:14: error: catching polymorphic type ‘struct dev::BadHexCharacter’ by value [-Werror=catch-value=]
  catch (dev::BadHexCharacter)
              ^~~~~~~~~~~~~~~
/mnt/data/repos/solidity/libsolidity/interface/StandardCompiler.cpp: In member function ‘Json::Value dev::solidity::StandardCompiler::compileInternal(const Json::Value&)’:
/mnt/data/repos/solidity/libsolidity/interface/StandardCompiler.cpp:369:16: error: catching polymorphic type ‘struct dev::BadHexCharacter’ by value [-Werror=catch-value=]
    catch (dev::BadHexCharacter)
                ^~~~~~~~~~~~~~~
```

Reply:
The real fix is to append `const&` to it.